nntrainer/graph/graph_node.h

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file	graph_node.h
+ * @date	1 April 2021
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug		No known bugs except for NYI items
+ * @brief	This is the graph node interface for c++ API
+ */
+
+#ifndef __GRAPH_NODE_H__
+#define __GRAPH_NODE_H__
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace nntrainer {
+
+/**
+ * @class   Layer Base class for the graph node
+ * @brief   Base class for all layers
+ */
+class GraphNode {
+public:
+  /**
+   * @brief     Destructor of Layer Class
+   */
+  virtual ~GraphNode() = default;
+
+  /**
+   * @brief     Get index of the node
+   *
+   */
+  virtual size_t getIndex() = 0;
+
+  /**
+   * @brief     Set index of the node
+   *
+   */
+  // virtual void setIndex(size_t) = 0;
+
+  /**
+   * @brief     Get the Name of the underlying object
+   *
+   * @return std::string Name of the underlying object
+   * @note name of each node in the graph must be unique
+   */
+  virtual std::string getName() noexcept = 0;
+
+  /**
+   * @brief     Set the Name of the underlying object
+   *
+   * @param[in] std::string Name for the underlying object
+   * @note name of each node in the graph must be unique
+   *
+   * @todo make it work with setProperty
+   */
+  // virtual int setName(std::string name) = 0;
```

Comment:
I guess this is intended to unify interface to set property

Reply:
Yes, but I am not sure if this can be surely eliminated. It will be removed if future PRs work without it.